src/module-elasticsuite-tracker/Api/Customer/ReconciliationServiceInterface.php

```diff
@@ -0,0 +1,62 @@
+<?php
+/**
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade Smile Searchandising Suite to newer
+ * versions in the future.
+ *
+ * @category  Smile
+ * @package   Smile\ElasticsuiteTracker
+ * @author    Romain Ruaud <romain.ruaud@smile.fr>
+ * @copyright 2018 Smile
+ * @license   Open Software License ("OSL") v. 3.0
+ */
+
+namespace Smile\ElasticsuiteTracker\Api\Customer;
+
+/**
+ * Customer Reconciliation : used to handle correlation between customer_id and session_id/visitor_id
+ *
+ * @category Smile
+ * @package  Smile\ElasticsuiteTracker
+ * @author   Romain Ruaud <romain.ruaud@smile.fr>
+ */
+interface ReconciliationServiceInterface
+{
+    /**
+     * Add a reconciliation tuple into the database.
+     *
+     * @param int    $customerId The customer Id.
+     * @param string $sessionId  The session Id.
+     * @param string $visitorId  The visitor Id.
+     *
+     * @return void
+     */
+    public function addByCustomerId(int $customerId, string $sessionId, string $visitorId);
```

Comment:
Replace by : addEvent($eventData)
* The event data is stored in the queue
* The reconciliation is saved if needed

We hide the implementation.
Replace the \Smile\ElasticsuiteTracker\Api\EventQueueInterface by our CustomerTrackingServiceInterface in the hit controller.